htlcswitch/link.go

```diff
@@ -373,8 +397,30 @@ type channelLink struct {
 	downstream chan *htlcPacket
 
 	// shutdownRequest is a channel that the channelLink will listen on to
-	// service shutdown requests from ShutdownIfChannelClean calls.
-	shutdownRequest chan *shutdownReq
+	// service shutdown requests from NotifyShouldShutdown calls.
+	shutdownRequest chan *chanCloseCtx
+
+	// localCloseReq stores the local ChanClose request to pass to the
+	// peer.Brontide when we're done with the Shutdown phase.
+	localCloseReq *ChanClose
+
+	// shutdownInit is a bool that is set when we've initiated a coop
+	// close.
+	shutdownInit bool
```

Comment:
Have we thought about giving the link a `state` and advancing its states? That probably could make things a bit easier and more abstracted.

Reply:
I think we should do that in a different PR